sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -955,9 +959,21 @@ <T> Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> queryItemsInternalFu
         CosmosAsyncClient client = this.getDatabase().getClient();
         CosmosQueryRequestOptions options =
             cosmosQueryRequestOptions != null ? cosmosQueryRequestOptions : new CosmosQueryRequestOptions();
-
+        
         Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> pagedFluxOptionsFluxFunction = (pagedFluxOptions -> {
             String spanName = this.queryItemsSpanName;
+            
+            ShowQueryOptions showQueryOptions = clientTelemetryConfigAccessor.showQueryOptions(client.getClientTelemetryConfig());
+
+            if (ShowQueryOptions.PARAMETERIZED_ONLY.equals(showQueryOptions)
+                    && sqlQuerySpec.getParameters() != null
+                    && sqlQuerySpec.getParameters().size() > 0) {
```

Comment:
I think this is a bit unintuitive. I would assume a cx would choose to allow traces for parameterized queries when an app exclusively uses the overrides with SqlQuerySpec vs. String. So, I think what needs to be done is to apps the state whether a query was created via string or SqlQuerySpec and use that here to evaluate whether to log. Makes sense?